contracts/Aavegotchi/facets/AavegotchiGameFacet.sol

```diff
@@ -328,4 +327,37 @@ contract AavegotchiGameFacet is Modifiers {
     function isAavegotchiLocked(uint256 _tokenId) external view returns (bool isLocked) {
         isLocked = s.aavegotchis[_tokenId].locked;
     }
+
+    ///@notice Allow the current owner/borrower of an NFT to reduce kinship while channelling alchemica
+    ///@dev will revert if the gotchi kinship is too low to channel or if the lending listing does not enable channeling
+    ///@param _gotchiId Id of the Gotchi used to channel
+    function reduceKinshipViaChanneling(uint32 _gotchiId) external {
+        //only realmDiamond can reduce kinship
+        require(msg.sender == s.realmAddress, "GotchiLending: Only Realm can reduce kinship via channeling");
+        //no need to do checks on _gotchiId since realmDiamond handles that
+        //first check if aavegotchi is lent
+        if (LibGotchiLending.isAavegotchiLent(_gotchiId)) {
+            //short-circuit here
+            uint32 listingId = s.aavegotchiToListingId[_gotchiId];
+            if (LibBitmapHelpers.getValueInByte(0, s.gotchiLendings[listingId].permissions) == 0) {
+                revert("This listing has no permissions set");
+            }
+
+            //check if channelling is allowed for the listing
+            //check that the modifier is at least 1
+            //more checks can be introduced if more modifiers are added
+            if (LibBitmapHelpers.getValueInByte(0, s.gotchiLendings[listingId].permissions) > 0) {
+                //more checks can be introduced here as different permissions are added
+                LibAavegotchi._reduceAavegotchiKinship(_gotchiId, 2);
```

Comment:
Is there an event emitted for this onchain status change? If not, the subgraph value for kinship will be incorrect. @Timidan